### ISSUE TITLE
When topologyAware is true getting failed to add OSD to crush map exit status 22

### ISSUE THREAD

**Issue by dsargent3220** [rook/rook#4247]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Attempting to create a new cluster with the topologyAware flag set to true and labels on the new nodes as outlined in the docs I get the error 
`failed to configure dirs map[/mnt/data01:4]. failed to initialize OSD at /mnt/data01/osd4: failed adding osd.4 to crush map: exit status 22`
On my 4 worker nodes I have the following labelsl
```
failure-domain.beta.kubernetes.io/region: pullman
failure-domain.beta.kubernetes.io/zone: pullman-3
rack: rack3
```
The rook-ceph-osd-prepare pods crash loop with the above error until they give up and no OSDs are created

Things I've tried to get it to work.

- Tried completing removing Rook/Ceph, including deleting the data directories on the nodes.  This is done for every test below
- Removed the `failure-domain.beta.kubernetes.io/region` label and rebuilt cluster
- Removed the `topologyAware: true` setting from the cluster.yaml file, this will cause the OSDs to create but the crush map then ignores the labels on the nodes and we end up with a standard node based crush map

**Expected behavior:**
Expect the crush map to be created as expected and the OSDs to be crated.

**How to reproduce it (minimal and precise):**

1.  Add labels as outlined above to each of your worker nodes
2.  Add teh Rook/Ceph helm chart
3.  Create a cluster with the following settings
```
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  cephVersion:
    image: ceph/ceph:v14.2.4-20190917
    allowUnsupported: false
  dataDirHostPath: /var/lib/rook
  skipUpgradeChecks: false
  mon:
    count: 3
    allowMultiplePerNode: false
  dashboard:
    enabled: true
    ssl: false
  monitoring:
    # requires Prometheus to be pre-installed
    enabled: true
    rulesNamespace: rook-ceph
  network:
    hostNetwork: false
  # The option to automatically remove OSDs that are out and are safe to destroy.
  removeOSDsIfOutAndSafeToRemove: true
  storage: # cluster level storage configuration and selection
    topologyAware: true
    useAllNodes: false
    useAllDevices: false
# Cluster level list of directories to use for filestore-based OSD storage. If uncomment, this example would create an OSD under the dataDirHostPath.      
    directories:
    - path: /mnt/data01
    nodes:
    - name: "lpul-k8slatestwrk1"
    - name: "lpul-k8slatestwrk2"
    - name: "lpul-k8slatestwrk3"
    - name: "lpul-k8slatestwrk4"
  disruptionManagement:
    managePodBudgets: true
    osdMaintenanceTimeout: 30
```

**Environment**:
* OS (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
* Kernel (e.g. `uname -a`):
`Linux lpul-k8slatestwrk4 3.10.0-1062.4.1.el7.x86_64 #1 SMP Fri Oct 18 17:15:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration:  
`VMWare, on prem`
* Rook version (use `rook version` inside of a Rook Pod):
```
rook: v1.1.4
go: go1.11
```
* Storage backend version (e.g. for ceph do `ceph -v`):
`ceph version 14.2.4 (75f4de193b3ea58512f204623e6c5a16e6c1e1ba) nautilus (stable)`
* Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T17:01:15Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.2", GitCommit:"c97fe5036ef3df2967d086711e6c0c405941e14b", GitTreeState:"clean", BuildDate:"2019-10-15T19:09:08Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
`Rancher`
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox]
`HEALTH_OK`

**Comment by leseb** [comment]
This will be fixed in https://github.com/rook/rook/pull/4131

**Comment by travisn** [comment]
the operator always uses topologyAware since the 1.2 release and I haven't heard of a repro of this issue, so must have been fixed by #4131
